src/main/java/org/hydev/mcpm/client/installer/InstallController.java

```diff
@@ -0,0 +1,86 @@
+package org.hydev.mcpm.client.installer;
+
+import org.hydev.mcpm.client.Downloader;
+import org.hydev.mcpm.client.database.DatabaseInteractor;
+import org.hydev.mcpm.client.database.LocalPluginTracker;
+import org.hydev.mcpm.client.database.boundary.SearchPackagesBoundary;
+import org.hydev.mcpm.client.database.fetcher.LocalDatabaseFetcher;
+import org.hydev.mcpm.client.database.inputs.SearchPackagesType;
+import org.hydev.mcpm.client.database.results.SearchPackagesResult;
+import org.hydev.mcpm.client.installer.input.InstallInput;
+import org.hydev.mcpm.client.models.PluginModel;
+
+import java.io.File;
+import java.net.URI;
+
+/**
+ * Install Controller which handles the install input and request the database to
+ * serve installing use case
+ */
+public class InstallController implements InstallBoundary {
+    private final InstallInteractor installInteractor;
+    private final SpigotPluginDownloader spigotPluginDownloader;
+    private final Downloader downloader;
+
+
+    public InstallController() {
+        this.installInteractor = new InstallInteractor();
+        this.downloader = new Downloader();
+        this.spigotPluginDownloader = new SpigotPluginDownloader(this.downloader);
+    }
+
+    /*
+    * Install the plugin
+    * @param installInput: the plugin submitted by the user for installing
+    * */
+    @Override
+    public void installPlugin(InstallInput installInput) {
+        SearchPackagesResult searchPackagesResult = installInteractor.searchPackageInput(installInput);
+
+        // Check if the searched plugin for installing is found
+        if (searchPackagesResult.plugins().isEmpty()) {
+            throw new RuntimeException();
+        }
+
+        // Pick the plugin with the latest version
+        PluginModel latestPluginModel = searchPackagesResult.plugins().get(0);
+        long id = latestPluginModel.id();
+        for (PluginModel plugin: searchPackagesResult.plugins()) {
+            if (plugin.id() > id) {
+                id = plugin.id();
+                latestPluginModel = plugin;
+            }
+        }
+        var pluginLatestVersion = latestPluginModel.getLatestPluginVersion();
+        if (pluginLatestVersion.isEmpty()) {
+            throw new RuntimeException();
```

Comment:
Might want to throw something more specific for debugging ease. Make a comment on the side for a remainder if you don;t want to deal with it for now.